### ISSUE TITLE
[powershell] `configuration` and `node` keywords aren't highlighted

### ISSUE THREAD

**Issue by pcgeek86** [Microsoft/vscode#3342]
On Mac OS X, the `configuration` keyword isn't being syntax highlighted for the PowerShell language. Is this an easy fix?

<img width="227" alt="screen shot 2016-02-23 at 8 43 24 am" src="https://cloud.githubusercontent.com/assets/466713/13256833/86f14374-da09-11e5-9d28-ebda618ab55b.png">

Same goes for `node` keyword.

<img width="276" alt="128ee370-da21-11e5-816a-329e47b35084" src="https://cloud.githubusercontent.com/assets/22350/13280142/949175e8-dadc-11e5-9a31-a4612f229e93.png">

Cheers,
Trevor Sullivan
Microsoft MVP: PowerShell

**Comment by joaomoreno** [comment]
This would be a great contribution! Would you like to [give it a try](https://github.com/Microsoft/vscode/tree/master/extensions/powershell) and submit a PR?

**Comment by pcgeek86** [comment]
@joaomoreno I don't think I'll be very efficient at fixing it, since I'm not on the dev team. It would take me quite a while to research the specifics about implementing syntax highlighting for VS Code, and then track down the problem.

My guess is that the problem lies somewhere here: https://github.com/Microsoft/vscode/blob/master/extensions/powershell/syntaxes/PowershellSyntax.tmLanguage#L525

**Comment by egamma** [comment]
FYI @daviwil

**Comment by joaomoreno** [comment]
Interesting. From your code snippet, it appears that `configuration` is indeed *captured* but the theme somehow is not colouring it. Is there any editor theme that colors it? You can toggle between them using the <kbd>F1</kdb>, `Change Theme` action.

ping @aeschli

**Comment by pcgeek86** [comment]
I talked to Keith Hill on Skype this morning, and he pointed out that this bug is resolved in an alpha build, that I don't currently have access to. I'll close this ticket unless things change after the February release.

Cheers,
Trevor Sullivan

**Comment by pcgeek86** [comment]
@joaomoreno Actually, you were right. I just tested out changing the theme, and sure enough, it is colored. I never changed the them from the default (didn't even know this was available), so maybe the theme needs to be updated? Re-opening issue.

**Comment by joaomoreno** [comment]
Awesome. @aeschli, what can be done about this?

**Comment by aeschli** [comment]
`configuration` seems to use the `storage.type` scope, which is a bit of a complicated case.
Some grammars use that for type references, other for keywords like 'var'.
Our default themes color it with the 'type' color which is turquise in the dark+/light+ and regular background color in the dark VS/light VS themes.

I have to read more on what's the expected use of `storage.type` and might change it in te future to a keyword color. But if you want to move foreard, better use a scope like `keyword'.

**Comment by egamma** [comment]
Actually, this was intentional, we did not want to change the selected theme behind the user's back. Consider users that like the less colorful theme, they would not appreciate if we upgrade them automatically.

However, given this issue I wonder whether we should provide some more active guidance to users to upgrade to the '+' theme (they obviously do not read the release notes :disappointed:.

One idea would be a 'once message' (shown only once) to a user that uses the Dark, Light them that there now is a Dark+/Light+. The message could provide a command to change the theme.

**Comment by pcgeek86** [comment]
@aeschli Thanks for the analysis. I'm not sure why the `storage.type` was used, because I don't understand custom grammar authorship for Visual Studio Code at this time.

@egamma Good idea to mention this in the release notes, somehow. *In fact, I didn't even realize that release notes were available. Normally I love reading them.* Perhaps we could do something to encourage reading release notes, like showing them in the editor after an upgrade or new installation of VS Code?

**Comment by egamma** [comment]
@pcgeek86 the release notes are here https://code.visualstudio.com/Updates :smile:  and whenever you get an auto update there is message box with a button to open the release notes. 

The fact that you never noticed them makes me thinking...

**Comment by bgse** [comment]
@egamma This might be a psychology thing. I guess when the update notification comes up, users can get a bit excited about new toys, and the obvious thing to do is to hit that update button asap.

Maybe it would be a good idea to display the release notes notification again once the user starts the updated VSCode for the first time.

E.g. "Visual Studio Code was recently updated. Do you want to read release notes now", offering options like "Release Notes", "Not now", "Do not ask again".

**Comment by pcgeek86** [comment]
@bgse @egamma omg yes, you're spot on with that analysis @bgse. I literally do that.

I still like my idea of displaying the release notes after an upgrade, or alternatively, showing **just** the release notes button / notification as @bgse suggested. :smile: :smile:  Maybe have an option to disable it in the VS Code config?

**Comment by egamma** [comment]
@bgse and @pcgeek86 thanks for your insights, we spend some real time on the release notes, it would be a pity if they are not getting read.

// @joaomoreno

**Comment by joaomoreno** [comment]
#3525

**Comment by pcgeek86** [comment]
@joaomoreno Thanks for filing that.

**Comment by aeschli** [comment]
@daviwil David, ok if we move this issue to https://github.com/PowerShell/vscode-powershell ? Once fixed we're happy to update our powershell text mate grammar.

**Comment by daviwil** [comment]
We're actually about to start maintaining the official PowerShell TextMate grammar in its own repo so you'll be able to move all PowerShell grammar bugs there. I'll see about getting that repo created today and I'll create a bug there to track this issue.

**Comment by daviwil** [comment]
This was fixed in the VS Code 1.3, let us know if you see any issues with these keywords in the future.  Thanks!

**Comment by aeschli** [comment]
![image](https://cloud.githubusercontent.com/assets/6461412/18202020/daae496c-710d-11e6-8959-ccfe378feca2.png)
'node' is not a separate token, and therefor doesn't get a different color. @daviwil is this a bug?

**Comment by pcgeek86** [comment]
@aeschli Yes, it would be a bug if it still exists, but @daviwil indicated it was fixed. Unfortunately it seems it's still a problem.

Here's what I see in VSCode:

![image](https://cloud.githubusercontent.com/assets/466713/18206439/f52f3cc6-70e3-11e6-8bf6-da201fb607aa.png)

Here's what I see in PowerShell ISE:

![image](https://cloud.githubusercontent.com/assets/466713/18206445/fb43c3a2-70e3-11e6-9210-693475ba852e.png)

**Comment by daviwil** [comment]
It was fixed in our new syntax definition in VS Code 1.3 but we had to pull it back out in 1.4 due to other bugs that were severely impacting user experience.  We're going to start fixing up the new syntax definition over at https://github.com/PowerShell/EditorSyntax and hopefully bring it back within the next couple of VS Code releases.
